source/d3d10/d3d10_device.cpp

```diff
@@ -246,6 +248,14 @@ void    STDMETHODCALLTYPE D3D10Device::OMSetRenderTargets(UINT NumViews, ID3D10R
 #if RESHADE_DX10_CAPTURE_DEPTH_BUFFERS
 	track_active_rendertargets(NumViews, ppRenderTargetViews, pDepthStencilView);
 #endif
+
+	if (_runtimes.empty())
+		return;
```

Comment:
Shouldn't return here, so the original `OMSetRenderTargets` is called even if no runtime is associated with this device (this can happen depending on the settings the device was created with).

This is better:
```c++
if (!_runtime.empty())
{
    const auto runtime = _runtimes.front();
    runtime->on_set_depthstencil_view(pDepthStencilView);
}
```

Reply:
Yes, you totally right. I will change this